### ISSUE TITLE
Access to mpd via trusted uplink.

### ISSUE THREAD

**Issue by nis65** [Daenou/ansible-multiroom-audio#69]
Usually, the uplink interface should be locked down as much as possible as the network connected there is out of our own control (e.g. a public access point).

However, when the uplink is connected to a trusted network (e.g. your home network) and you can connect the mpd clients to that same network, direct access to the mpd ports is needed via uplink.

The implementation should be controlled by a Setting in a file, e.g.

```
UPLINK_TRUSTED=yes
```

**Comment by nis65** [comment]
When we are connected to a trusted uplink (via Ethernet or External USB WLAN Dongle),
* the uplink should accept mdns (UCP src/dest port 5353) traffic
* the uplink should accept mpd traffic (tcp port 6600)
* if not needed, the WLAN AP should be disabled (at least rfkill, but hostapd and dnsmasq could be switched off too) to free up the frequency.

**Comment by nis65** [comment]
the current mpd config (raspbian default) with enabled mpd.socket does not allow to use mDNS (zeroconf). Enabled zeroconf for test purposes manually by adding two lines to `mpd.conf`:

~~~
zeroconf_enabled               "yes"
zeroconf_name                  "MPD @ %h"
~~~

and disabling the systemd `mpd.socket` (does stop make sense?):

~~~
systemctl stop mpd.socket
systemctl disable mpd.socket
~~~

**Comment by nis65** [comment]
Talking to myself: Well, this is not that a good idea after all. What I want to achieve is to use my standalone camper Pi in the same way as at home (i.e. a plain music device without any routing/secuity requirements. In other words: *Not apply the `accesspoint` and `uplink` roles.

So what is left from that feature request (at the moment)
* A script to temporarily disable all features from the `accesspoint` and `uplink` roles. It would be nice if `ap0` (while not in use by `hostapd`) could be used again as normal uplink.
* activate zeroconf. Now that I invested a little time into understanding it, this seems to work really well and besides being a bit noisy on the network relieves the end users from configuring a DNS server...

**Comment by nis65** [comment]
Manual stopping is solved (previous comment).

The additional issue (zeroconf) is solved with #71.
